proposals/dictionary-expressions.md

```diff
@@ -165,47 +165,52 @@ From [Open Question 3](#open-question-3), we will support *dictionary elements*
->
-> - Array rules...
-> - Span rules...
->
-> - A type with a create method with an iteration type determined from a GetEnumerator instance method or enumerable interface, not from an extension method.
-> - ```diff
->   + A type with a create method with an iteration type determined
->   + from a GetEnumerator instance method or enumerable interface,
->   + not from an extension method, that is some `KeyValuePair<TKey, TValue>`
->   + and an argument type `IEnumerable<KeyValuePair<TKey, TValue>`.
-> 
->   + For example:
->   + `public static ImmutableDictionary CreateRange<TKey, TValue>(IEnumerable<KeyValuePair<TKey, TValue>>)`.
->   + Note: it is an open question what collection types are supported for the argument type.
->   ```
-> - A struct or class type that implements System.Collections.IEnumerable where:
->   - The type has an applicable constructor that can be invoked with no arguments, and the constructor is accessible at the location of the collection expression.
->   - If the collection expression has any elements, the type has an applicable instance or extension method Add that can be invoked with a single argument of the iteration type, and the method is accessible at the location of the collection expression.
->   - ```diff
->     + If the collection expression has any elements and the type 
->     + has an iteration type of some `KeyValuePair<TKey, TValue>`
->     + and the type has applicable indexer that can be invoked with
->     + a single argument of the `TKey` type, and a value of the `TValue`
->     + type, and the indexer is accessible at the location of the
->     + collection expression.
->     ```
-> - An interface type:
->   - `System.Collections.Generic.IEnumerable<T>`
->   - `System.Collections.Generic.IReadOnlyCollection<T>`
->   - `System.Collections.Generic.IReadOnlyList<T>`
->   - `System.Collections.Generic.ICollection<T>`
->   - `System.Collections.Generic.IList<T>`
->   - ```diff
->     + System.Collections.Generic.IDictionary<TKey, TValue>
->     + System.Collections.Generic.IReadOnlyDictionary<TKey, TValue>
->     ```
+*Collection expression conversions* are updated to include conversions to *dictionary types*.
+
+An implicit *collection expression conversion* exists from a collection expression to the following *dictionary types*:
+* A *dictionary type* with an appropriate *[create method](#create-methods)*.
+* A *struct* or *class* *dictionary type* that implements `System.Collections.IEnumerable` where:
+  * The *element type* is determined from a `GetEnumerator` instance method or enumerable interface.
+  * The *type* has an *[applicable](https://github.com/dotnet/csharpstandard/blob/standard-v6/standard/expressions.md#11642-applicable-function-member)* constructor that can be invoked with no arguments, and the constructor is accessible at the location of the collection expression.
+  * The *indexer* has a setter that is as accessible as the declaring type.
```

Comment:
nice.  i like that we only need to add this rule about the indexer setter here.  that seems very intuitive as we're basically saying "We have to actually new this up, and then actually set the values".